src/System.Xml.XmlSerializer/src/System/Xml/Serialization/Compilation.cs

```diff
@@ -305,21 +303,27 @@ public override int GetHashCode()
 
     internal class TempAssemblyCache
     {
-        private Hashtable _cache = new Hashtable();
+        private Dictionary<TempAssemblyCacheKey, TempAssembly> _cache = new Dictionary<TempAssemblyCacheKey, TempAssembly>();
```

Comment:
I was under the impression that Hashtable was used intentionally as well but it turns out it is a bit confusing that all of the ```Hashtable``` usage was overridden with ```InternalHashtable``` in ```using Hashtable = System.Collections.InternalHashtable;```, which in turn was a derivative of Dictionary<object, object>. It seems that there is a meaningful context when this overridden behavior was introduced. However this now looks like a shorthand to Dictionary<object, object> which should be removed.